src/nvim/api/win_config.c

```diff
@@ -409,23 +409,28 @@ void nvim_win_set_config(Window window, Dict(win_config) *config, Error *err)
   if (!parse_win_config(win, config, &fconfig, !was_split || to_split, err)) {
     return;
   }
+  win_T *parent = NULL;
+  if (config->win != -1) {
+    parent = find_window_by_handle(fconfig.window, err);
+    if (!parent) {
+      return;
+    } else if (to_split && parent->w_floating) {
+      api_set_error(err, kErrorTypeException, "Cannot split a floating window");
+      return;
+    }
+
+    // Prevent autocmd window from being moved into another tabpage
+    if (is_aucmd_win(win) && win_find_tabpage(win) != win_find_tabpage(parent)) {
```

Comment:
@zeertzjq wasn't totally sure about the autocmd win (didn't know about it at all before this tbh). Is just checking and throwing an error a reasonable way to ensure it's not moved to another tabpage by `set_config`?

Reply:
Yes

Reply:
Awesome, thanks!